### DISCUSSION TITLE
Launch Ubuntu on Docker not working

### DISCUSSION THREAD

**Discussion by subkaitaku** [nextcloud/all-in-one#4010] [Questions] [answered]
Hello.
I would like to launch NextCloud on my on-premise server(Ubuntu 22.04.3 LTS (GNU/Linux 6.2.0-39-generic x86_64)) via Cloudflare Tunnel, and I did so with the following reference, but it is a bad gateway without any error.
I don't think the request is reaching the destination, is there anything else to check?
https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md

## Run Command
- setted APACHE_PORT/SKIP_DOMAIN_VALIDATION /APACHE_IP_BINDING
- disabled Rocket Loader 

```bash
sudo docker run \
--sig-proxy=false \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 8080:8080 \
--env APACHE_PORT=11000 \
--env APACHE_IP_BINDING=0.0.0.0 \
--env SKIP_DOMAIN_VALIDATION=true \
--env NEXTCLOUD_DATADIR="/mnt/nextcloud" \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
--add-host=host.docker.internal:host-gateway \
nextcloud/all-in-one:latest

Trying to fix docker.sock permissions internally...
Adding internal www-data to group ping
Initial startup of Nextcloud All-in-One complete!
You should be able to open the Nextcloud AIO Interface now on port 8080 of this server!
E.g. https://internal.ip.of.this.server:8080

If your server has port 80 and 8443 open and you point a domain to your server, you can get a valid certificate automatically by opening the Nextcloud AIO Interface via:
https://your-domain-that-points-to-this-server.tld:8443
{"level":"info","ts":1704671759.828878,"msg":"using provided configuration","config_file":"/Caddyfile","config_adapter":""}
[07-Jan-2024 23:55:59] NOTICE: fpm is running, pid 136
[07-Jan-2024 23:55:59] NOTICE: ready to handle connections
[Sun Jan 07 23:55:59.849825 2024] [mpm_event:notice] [pid 131:tid 139622078528328] AH00489: Apache/2.4.58 (Unix) OpenSSL/3.1.4 configured -- resuming normal operations
[Sun Jan 07 23:55:59.849871 2024] [core:notice] [pid 131:tid 139622078528328] AH00094: Command line: 'httpd -D FOREGROUND
```

Access `strage.example.com`
<img width="654" alt="スクリーンショット 2024-01-08 9 24 49" src="https://github.com/nextcloud/all-in-one/assets/113927971/141240e9-4b73-4adf-ac47-46d792fdd6e4">

<img width="887" alt="image" src="https://github.com/nextcloud/all-in-one/assets/113927971/6c8fc438-ade1-4a30-9960-ffc0c965dbd9">

I tried  `host.docker.internal:11000`, but it didn't solve the problem.

## Curl on internal
```
curl http://***************:8080
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>400 Bad Request</title>
</head><body>
<h1>Bad Request</h1>
<p>Your browser sent a request that this server could not understand.<br />
Reason: You're speaking plain HTTP to an SSL-enabled server port.<br />
 Instead use the HTTPS scheme to access this URL, please.<br />
</p>
<hr>
<address>Apache/2.4.58 (Unix) Server at localhost Port 8080</address>
</body></html>
```

## Confirm Firewall Status
```bash
sudo ufw status
Status: inactive
```

**Comment by szaimen** [comment]
Hi, can you follow https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#6-how-to-debug-things?

**Comment by Zoey2936** [comment, marked as answer]
> You're speaking plain HTTP to an SSL-enabled server port.

try https://****:8080 instead of http:

**Comment by subkaitaku** [comment]
tred this.
```
$ curl https://localhost:8080
curl: (60) SSL certificate problem: self-signed certificate
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.

$ curl --insecure https://localhost:8080
redireced
```

I tried to access `https://localhost:8080` from Cloudflare tunnnel via subdomain but got 500 error

**Comment by subkaitaku** [comment]
I could not arrive with my self-certificate.

solved:
check this
<img width="1276" alt="image" src="https://github.com/nextcloud/all-in-one/assets/113927971/50416190-f4d2-458e-9f94-55a0a1ab282f">

to:
<img width="635" alt="image" src="https://github.com/nextcloud/all-in-one/assets/113927971/9264d8cb-cd83-4a4a-89b9-3d894ca6e7f6">
